src/test/scala/com/microsoft/hyperspace/index/rules/JoinIndexRuleTest.scala

```diff
@@ -320,6 +332,23 @@ class JoinIndexRuleTest extends HyperspaceRuleTestSuite with SQLHelper {
     }
   }
 
+  test("Join rule doesn't update plan if columns don't have one-to-one mapping case insensitive") {
+    val t1ProjectNode = Project(Seq(t1c1, t1c3), t1FilterNode)
+    val t2ProjectNode = Project(Seq(t2c1, t2c3), t2FilterNode)
+
+    val t1c1Caps = t1c1.withName("T1C1")
+    val t2c1Caps = t2c1.withName("T2C1")
+
+    val joinCondition = And(EqualTo(t1c1, t2c1), EqualTo(t1c1Caps, t2c1Caps))
+    val originalPlan = Join(t1ProjectNode, t2ProjectNode, JoinType("inner"), Some(joinCondition))
+    val updatedPlan = JoinIndexRule(originalPlan)
+    assert(!updatedPlan.equals(originalPlan))
```

Comment:
this contradicts with the test description?

Reply:
thanks @imback82 , fixed